### PR TITLE
feat: staging lambda deploy and test workflows

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add GitHub Workflows to build, push, deploy and test the Staging
lambda functions.

Also cleans up the CI Docker build workflow to remove the
AWS login.

# Note
I'll keep this as draft until Staging is fully operational.

### CODE REVIEW

File: .github/workflows/build_and_push_staging.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        image: ${{ fromJSON(needs.changes.outputs.images) }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+
+      - name: Configure AWS credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@05b148adc31e091bafbaf404f745055d4d3bc9d2 # tag=v1.6.1
+        with:
+          role-to-assume: arn:aws:iam::806545929748:role/OIDCGithubWorkflowRole
```
Comment (mohamed-cds) [inline .github/workflows/build_and_push_staging.yml:52]: this is the prod account number. staging is 127893201980

File: .github/workflows/integration_test_staging.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+
+      - name: Configure AWS credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@05b148adc31e091bafbaf404f745055d4d3bc9d2 # tag=v1.6.1
+        with:
+          role-to-assume: arn:aws:iam::806545929748:role/OIDCGithubWorkflowRole
```
Comment (mohamed-cds) [inline .github/workflows/integration_test_staging.yml:34]: same here about account number -> `127893201980`

**Review by mohamed-cds — APPROVED** [review]
LGTM other than swapping in new account number 🚀 ✅

File: .github/workflows/integration_test_staging.yml (hunk 2 above)
Comment (patheard) [reply .github/workflows/integration_test_staging.yml]: oooo yeah, good catch

**Pull request merged**